### PR TITLE
orderbook section updates

### LINKED ISSUE #484: Add orderbook list UI components
order book is very useful for more detailed inspection of bids
I propose to add an orderbook to the main bidding interface
initial draft

Order book examples
Binance:

dydx:

### PR DESCRIPTION
**Pull request by maria-vslvn**
Closes #484 

Logic should be implemented on #527

1. OrderbookContaner was created as wrapper for the orderbook components.
2. ButtonToggle component created as a new kind of buttons.
3. "Granularity" dropdown created with mockdata. The dropdown was made visible only for the list view.
4. OrderbookTable component created with mockdata. Table made responsive and scrollable.
5. Tooltips overlapping bug fixed for existing cases.
6. A few adjustments of the orderbook section were made for mobile view.
7. CSSType package was added for usage in styled components.
8. ChartIcon and TableIcon were added as components.

### CODE REVIEW

File: src/components/auction/OrderPlacement/index.tsx [hunk 1]
```diff
@@ -344,6 +358,15 @@ const OrderPlacement: React.FC<OrderPlacementProps> = (props) => {
 
   return (
     <>
+      <Wrap>
+        <SectionTitle as="h2">
+          {(auctionState === AuctionState.ORDER_PLACING ||
+            auctionState === AuctionState.ORDER_PLACING_AND_CANCELING) &&
+            'Place Order'}
+          {auctionState === AuctionState.CLAIMING && 'Claim Proceeds'}
```
Comment (josojo) [inline src/components/auction/OrderPlacement/index.tsx:366]: This component will never be rendered in the claiming state. No need for the distinguishment here.

Comment (josojo) [reply src/components/auction/OrderPlacement/index.tsx:366]: Indeed this logic needs to move into the claimer component

File: src/components/auction/OrderbookContainer/index.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+            activate={isChartVisible}
+            left={{ icon: <ChartIcon />, label: 'Chart' }}
+            onClick={() => setChartVisibility(!isChartVisible)}
+            right={{ icon: <TableIcon />, label: 'List' }}
+          />
+        </Wrap>
+      </Wrap>
+      {auctionState === AuctionState.PRICE_SUBMISSION && (
+        <AuctionPending>Auction closed. Pending on-chain price-calculation.</AuctionPending>
+      )}
```
Comment (josojo) [inline src/components/auction/OrderbookContainer/index.tsx:158]: This information used to be on the "claimer side" and not on the" orderbookcontainer side". Do you think it makes more sense on the orderbook side?

File: src/components/auction/OrderbookTable/index.tsx [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+          </TableCell>
+          <TableCell minWidth={'150px'}>
+            <Wrap>
+              <Wrap margin={'0 10px 0 0'}>Amount (GNO)</Wrap>
+              <Tooltip id="info" place={'bottom'} text={`You have no orders for this auction.`} />
+            </Wrap>
+          </TableCell>
+          <TableCell minWidth={'120px'}>
+            <Wrap>
+              <Wrap margin={'0 10px 0 0'}>Summary</Wrap>
```
Comment (josojo) [inline src/components/auction/OrderbookTable/index.tsx:201]: Nit: I think this was also wrong in the design. It should be just "sum" and not "summary". This field will represent the sum of all place orders up to that price

**Review by josojo — COMMENTED** [review]
Very nice! 

Quite close to the handover for the logic developer. Cool! Just here and there some nits.

File: src/components/auction/OrderPlacement/index.tsx (hunk 1 above)
Comment (maria-vslvn) [reply src/components/auction/OrderPlacement/index.tsx]: done

File: src/components/auction/OrderbookContainer/index.tsx (hunk 2 above)
Comment (maria-vslvn) [reply src/components/auction/OrderbookContainer/index.tsx]: done

File: src/components/auction/OrderbookTable/index.tsx (hunk 3 above)
Comment (maria-vslvn) [reply src/components/auction/OrderbookTable/index.tsx]: done

**Comment by gabitoesmiapodo** [comment]
Solved some conflicts, fixed a wonky margin.

Logic should be implemented on #527

File: src/components/buttons/buttonStylingTypes.ts [hunk 4]
```diff
@@ -14,6 +14,17 @@ export interface ButtonCommonProps {
 
 export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement>, ButtonCommonProps {}
 
+export interface ButtonGroupCommonProps {
+  left?: any
+  right?: any
```
Comment (ramirotw) [inline src/components/buttons/buttonStylingTypes.ts:19]: We should add the typing for left and right, what do you think?

```
{ icon: React.ReactElement, label: string }
```

**Comment by elena-zh** [comment]
Hi @maria-vslvn , @josojo , 
I have reviewed the PR. Changes LGTM besides several notes: 
1. It is impossible to select an option in the "Granularity" dropdown. I'm not sure, whether it should be implemented here or in #527 , but anyways I was not able to select an option there. Each time when I clicked on an option in the list, the dropdown was closed. 
2. Despite of the fact that Gabito mentioned in the https://github.com/gnosis/ido-ux/issues/484#issuecomment-830219487 , that there is no need to style the scrollbar, I think that it looks ugly and needs to be designed anyway. @josojo , what do you think about it?
scrols

3. It would be nice to add a padding between the last value and a scrollbar
image

Thanks

**Comment by josojo** [comment]
@elena-zh 
yes, you are right! Thanks for pushing the quality!

@maria-vslvn would you mind changing the scrollbar to something that fits well with the background?

**Comment by maria-vslvn** [comment]
@josojo @elena-zh pr updated

**Comment by josojo** [comment]
> @josojo @elena-zh pr updated

Thanks!
yes, the shape is looking much better. 
Do you think with a blue scrollbar it would look even better?

I am unsure

**Comment by elena-zh** [comment]
Hey @maria-vslvn , 
I do not see any changes on my side. 
Could you please take a look into the last commit? It seems that it is not pushed to the branch. 
Thanks

**Comment by ramirotw** [comment]
@elena-zh  the preview app was not updated as the build failed (@maria-vslvn added a new dependency and the yarn.lock file was not updated)

**Comment by maria-vslvn** [comment]
@elena-zh yarn.lock is updated

**Comment by elena-zh** [comment]
@maria-vslvn , vertical scroll looks great to me. 
However, horizontal scrollbar is still unchanged. Could you please fix it?
image

Thanks

**Comment by maria-vslvn** [comment]
@elena-zh could you please tell me how do you get the scrollbar? cause i haven't got any scrollbars, i mean: resolution, browser or smt else?

**Comment by elena-zh** [comment]
@maria-vslvn , 
when a screen width is less than 580 px
image

**Comment by maria-vslvn** [comment]
@elena-zh but on mobiles we have different scrollbars, so i didn't use the plugin on small screens for these cases, should i use it?
photo_2021-06-07_17-58-46

**Comment by elena-zh** [comment]
@maria-vslvn , I even do not know what to answer here. From my understanding all elements across the app should look more or less similar. I opened another table - Auctions list - for comparison to see how it looks like on such screen resolution and there is no horizontal scroll bar:
image

That is why I decided to indicate this case in this issue. 
Anyways, if it takes a lot of time to implement, we can leave it as it is. What do you think, @josojo ?

**Comment by elena-zh** [comment]
@maria-vslvn , 
I have created  low priority enhancement #654  for this horizontal scrollbar so we can fix it later (if needed). 
For now we can merge changes under the current pr into dev as all the rest changes LGTM.

Thanks

**Comment by ramirotw** [comment]
@maria-vslvn I've changed the base branch so you can squash&merge this PR. After integrating everything we will merge into develop.

**Comment by alongoni** [comment]
hey @maria-vslvn I found this small issue with the tooltips.
https://www.loom.com/share/ffb666930f0244dfbda84eff056c6bba

**Pull request merged**